src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -0,0 +1,261 @@
+          topic.partitions.begin(),
+          topic.partitions.end(),
+          std::back_inserter(resp.data.responses),
+          topic_response,
+          all_node_ids);
+
+        for (auto it = topic.partitions.begin(); it != valid_partitions_end;
+             ++it) {
+            const reassignable_partition& partition{*it};
+            model::ntp ntp{
+              model::kafka_namespace, topic.name, partition.partition_index};
+
+            if (partition.replicas.has_value()) {
+                vlog(
+                  klog.debug,
+                  "Request to reassign partitions: ntp {}, replicas {}",
+                  ntp,
+                  *partition.replicas);
+
+                auto errc
+                  = co_await ctx.topics_frontend().move_partition_replicas(
+                    ntp,
+                    std::move(*partition.replicas),
+                    model::timeout_clock::now() + request.data.timeout_ms);
+
+                if (!errc) {
+                    topic_response.partitions.push_back(
+                      reassignable_partition_response{
+                        .partition_index = partition.partition_index});
+                } else {
+                    auto clerr = static_cast<cluster::errc>(errc.value());
+                    vlog(
+                      klog.debug,
+                      "Failed to move partition replicas: ntp {}, ec {}",
+                      ntp,
+                      clerr);
+                    auto kerr = map_topic_error_code(clerr);
+                    topic_response.partitions.push_back(
+                      reassignable_partition_response{
+                        .partition_index = partition.partition_index,
+                        .error_code = kerr,
+                        .error_message = ss::sstring{error_code_to_str(kerr)}});
+                }
+
+            } else {
+                // Otherwise we cancel the pending request
+                vlog(
+                  klog.debug, "Request to cancel pending request: ntp {}", ntp);
+                auto errc = co_await ctx.topics_frontend()
+                              .cancel_moving_partition_replicas(
+                                ntp,
+                                model::timeout_clock::now()
+                                  + request.data.timeout_ms);
+                if (errc) {
+                    auto clerr = static_cast<cluster::errc>(errc.value());
+                    vlog(
+                      klog.debug,
+                      "Failed to cancel pending request: ntp {}, ec {}",
+                      ntp,
+                      clerr);
+                    // Explicitly check for no_update_in_progress here. We could
+                    // change the mapping in map_topic_error_code but that may
+                    // cause a regression since several places call this
+                    // function
+                    auto kerr = clerr == cluster::errc::no_update_in_progress
```

Comment:
I think this makes sense too, I agree with your comment. But maybe there's a nice solution, keeping in mind for future work

Reply:
I think the simple solution is to add `no_update_in_progress` to `map_topic_error_code` and then run all our tests repeatedly to test for regressions. I'll do this when I add waay more testing.

Reply:
Added this comment in followup ticket here https://github.com/redpanda-data/redpanda/issues/8325